app/src/androidTest/java/com/maxcraft/stackcalculator/MainActivityTest.java

```diff
@@ -0,0 +1,308 @@
+package com.maxcraft.stackcalculator;
+
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.suitebuilder.annotation.LargeTest;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.*;
+import static android.support.test.espresso.assertion.ViewAssertions.*;
+import static android.support.test.espresso.matcher.ViewMatchers.*;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Test
+    public void CleanBtnClickTest() {
+        ViewInteraction appCompatButton = onView(
+                allOf(withId(R.id.btn_clear), withText("clear"),
+                        childAtPosition(
+                                childAtPosition(
+                                        withClassName(is("android.widget.LinearLayout")),
+                                        1),
+                                0),
+                        isDisplayed()));
+        appCompatButton.perform(click());
+        onView(withId(R.id.indicator_txt)).check(matches(withText("0")));
+    }
+
+    @Test
+    public void pushBtnTestClick() {
+        ViewInteraction appCompatButton = onView(
+                allOf(withId(R.id.btn_push), withText("push"),
```

Comment:
Are you sure each field will always have the text you expect? It might be changed with translations.